### PR TITLE
bin: allow password-less sudo for install-requirements script

### PR DESCRIPTION
**Pull request by AlbinB97**
> [!warning]
> **This is a public repository, ensure not to disclose:**
>
> - [x] personal data beyond what is necessary for interacting with this pull request, nor
> - [x] business confidential information, such as customer names.

### What kind of PR is this?

**Required**: Mark one of the following that is applicable:

- [ ] kind/feature       
- [x] kind/improvement   
- [ ] kind/deprecation   
- [ ] kind/documentation 
- [ ] kind/clean-up      
- [ ] kind/bug           
- [ ] kind/other         

*Optional*: Mark one or more of the following that are applicable:

> [!important]
> Breaking changes should be marked `kind/admin-change` or `kind/dev-change` depending on type
> Critical security fixes should be marked with `kind/security`

- [ ] kind/admin-change 
- [ ] kind/dev-change   
- [ ] kind/security     
- [ ] [kind/adr]()      

### What does this PR do / why do we need this PR?

Allows you to install-requirements without password for environments that has password-less sudo. All you need to do is use the --no-pass flag.

...

#### Checklist

- [x] Proper commit message prefix on all commits
  
- Change checks:
  - [x] The change is transparent
  - [ ] The change is disruptive
  - [x] The change requires no migration steps
  - [ ] The change requires migration steps
  - [ ] The change upgrades CRDs
  - [ ] The change updates the config *and* the schema
- Metrics checks:
  - [ ] The metrics are still exposed and present in Grafana after the change
  - [ ] The metrics names didn't change (Grafana dashboards and Prometheus alerts are not affected)
  - [ ] The metrics names did change (Grafana dashboards and Prometheus alerts were fixed)
- Logs checks:
  - [ ] The logs do not show any errors after the change
- Pod Security Policy checks:
  - [ ] Any changed pod is covered by Pod Security Admission
  - [ ] Any changed pod is covered by Gatekeeper Pod Security Policies
  - [ ] The change does not cause any pods to be blocked by Pod Security Admission or Policies
- Network Policy checks:
  - [ ] Any changed pod is covered by Network Policies
  - [ ] The change does not cause any dropped packets in the `NetworkPolicy Dashboard`
- Audit checks:
  - [ ] The change does not cause any unnecessary Kubernetes audit events
  - [ ] The change requires changes to Kubernetes audit policy
- Falco checks:
  - [ ] The change does not cause any alerts to be generated by Falco
- Bug checks:
  - [ ] The bug fix is covered by regression tests

### CODE REVIEW

File: bin/install-requirements.bash [hunk 1]
```diff
@@ -7,9 +7,13 @@ export ANSIBLE_STDOUT_CALLBACK=yaml
 echo "This script will install all requirements"
 echo "Enter your password to start the install"
 
-if [ "${#}" -ge 2 ] && [ "${2}" = "--user" ]; then
+if [ "${#}" -ge 1 ] && [ "${1}" = "--user" ]; then
```
Comment (simonklb) [inline bin/install-requirements.bash:10]: This is incorrect if it's executed through `bin/ck8s`. If you want to keep this then you need to `shift` there as well.

File: bin/install-requirements.bash [hunk 2]
```diff
@@ -7,9 +7,13 @@ export ANSIBLE_STDOUT_CALLBACK=yaml
...
   export CK8S_INSTALL_PATH="${CK8S_INSTALL_PATH:-"${HOME}/bin"}"
   ansible-playbook -e 'ansible_python_interpreter=/usr/bin/python3' --ask-become-pass roles/get-requirements.yaml
 else
-  ansible-playbook -e 'ansible_python_interpreter=/usr/bin/python3' --ask-become-pass --become --become-user root roles/get-requirements.yaml
+  if [ "${#}" -ge 1 ] && [ "${1}" = "--become" ]; then
```
Comment (simonklb) [inline bin/install-requirements.bash:14]: What about `--no-ask-become-pass`? `--become` already has a meaning.

**Review by simonklb — CHANGES_REQUESTED** [review]
Please update the `ck8s` usage as well!

File: bin/install-requirements.bash (hunk 1 above)
Comment (AlbinB97) [reply bin/install-requirements.bash:10]: ah ye, that makes sense. Would a case statement with a loop make more sense here then maybe? Then both `bin/ck8s` and just running the script individually would work. So this script itself just checks for any arguments and if they are equal to the expected ones it does the right thing.

Comment (AlbinB97) [reply bin/install-requirements.bash:10]: yeah I guess, I went with just adding `shift` to `bin/ck8s` and didn't make a case statement. PTAL 😄

File: bin/install-requirements.bash (hunk 2 above)
Comment (AlbinB97) [reply bin/install-requirements.bash]: changed this to `--no-pass` short and sweet

Comment (simonklb) [reply bin/install-requirements.bash]: Nice! :)

**Review by simonklb — APPROVED** [review]

**Pull request merged**